entc/integration/integration_test.go

```diff
@@ -1406,11 +1406,46 @@ func Tx(t *testing.T, client *ent.Client) {
 		var m mocker
 		m.On("onRollback", nil).Once()
 		defer m.AssertExpectations(t)
-		tx.OnRollback(m.rHook())
+
+		tx.OnRollback(func(next ent.Rollbacker) ent.Rollbacker {
+			return ent.RollbackFunc(func(ctx context.Context, tx *ent.Tx) error {
+				err := next.Rollback(ctx, tx)
+				m.onRollback(err)
+
+				require.NotNil(t, ctx)
+
+				return err
+			})
+		})
 		err = tx.Item.Create().Exec(ctx)
-		require.Error(t, err)
+		require.Error(t, err) // because of read-only in tx opts
 		require.NoError(t, tx.Rollback())
 	})
+
+	t.Run("TxOptions Commit", func(t *testing.T) {
+		if client.Dialect() == dialect.SQLite {
+			t.Skip("Skipping SQLite")
+		}
+		tx, err := client.BeginTx(ctx, &sql.TxOptions{Isolation: stdsql.LevelReadCommitted})
+		require.NoError(t, err)
+		var m mocker
+		m.On("onCommit", nil).Once()
+		defer m.AssertExpectations(t)
+
+		tx.OnCommit(func(next ent.Committer) ent.Committer {
+			return ent.CommitFunc(func(ctx context.Context, tx *ent.Tx) error {
+				err := next.Commit(ctx, tx)
+				m.onCommit(err)
+				require.NotNil(t, ctx)
+
```

Comment:
```suggestion
```

Reply:
1439 line actually test changes, right? didn't get this suggestion

Reply:
He wants you to remove the whitespace / blank line ;-)

Reply:
Exactly. Thanks @masseelch

Reply:
Thanks pals, now I got it. No blank lines, sorry, didn't notice 

(honestly was supposing that, while looking on code-base. If you would like to add something like linter or style-guider it would be cool)